library/common/main_interface.cc

```diff
@@ -10,30 +10,39 @@
 
 // NOLINT(namespace-envoy)
 
-static std::shared_ptr<Envoy::Engine> strong_engine_;
-static std::weak_ptr<Envoy::Engine> engine_;
 static std::atomic<envoy_stream_t> current_stream_handle_{0};
 static std::atomic<envoy_network_t> preferred_network_{ENVOY_NET_GENERIC};
 
+static Envoy::Engine* engine(envoy_engine_callbacks callbacks = {0}) {
+  static Envoy::Engine* engine = new Envoy::Engine{callbacks, preferred_network_};
+  return engine; 
+}
+
 envoy_stream_t init_stream(envoy_engine_t) { return current_stream_handle_++; }
 
 envoy_status_t start_stream(envoy_stream_t stream, envoy_http_callbacks callbacks) {
-  if (auto e = engine_.lock()) {
```

Comment:
Really nice that we can remove these locks now

Reply:
:)